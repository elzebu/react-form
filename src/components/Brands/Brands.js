import React, { useState } from 'react'
import PropTypes from 'prop-types'

import Brand from '../Brand/Brand'

function Brands({ brands, click }) {

    const [displayBrand, setDisplayBrand] = useState(false)
    
    const list = brands.map(brand => (
        <Brand data={brand} key={brand.id} click={() => click(brand.id)} />
    ));
    return (
        <>
        <button onClick={() => setDisplayBrand(!displayBrand)}>
            {!displayBrand ?'Afficher': 'Masquer'} les marques
        </button>
          {displayBrand ?
            <>
                <h2>Liste des marques</h2>
                {list}
            </>
            : null }
        </>
    )
}

Brands.propTypes = {
    brands: PropTypes.array.isRequired,
    click: PropTypes.func.isRequired
}

export default Brands;
