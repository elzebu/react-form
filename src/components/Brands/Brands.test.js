import React from 'react'
import { create } from 'react-test-renderer'

import Brands from './Brands'


describe('<Brands />', () => {
    
    it('matches previous snapshot', () => {
        const component = create(
            <Brands brands={[]} click={() => {}} />,
        )
        let brand = component.toJSON()
        expect(brand).toMatchSnapshot()
    })

})

