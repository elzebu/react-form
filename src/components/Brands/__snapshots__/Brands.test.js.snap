// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brands /> matches previous snapshot 1`] = `
<button
  onClick={[Function]}
>
  Afficher
   les marques
</button>
`;

exports[`matches previous snapshot 1`] = `
<button
  onClick={[Function]}
>
  Afficher
   les marques
</button>
`;
